cmd/openshift-install/migrate.go

```diff
@@ -0,0 +1,186 @@
+	cmd.PersistentFlags().StringVar(&azureMigrateListOpts.privateZone, "privateZone", "", "Show a private zone and records")
+	cmd.PersistentFlags().StringVar(&azureMigrateListOpts.resourceGroup, "resourceGroup", "", "The resource group for the zone")
+	cmd.PersistentFlags().BoolVar(&azureMigrateListOpts.all, "all", false, "Show all zones and records")
+	cmd.PersistentFlags().BoolVar(&azureMigrateListOpts.public, "public", false, "Show public legacy zones")
+
+	return cmd
+}
+
+var (
+	azureMigrateOpts struct {
+		zone              string
+		oldResourceGroup  string
+		newResourceGroup  string
+		virtualNetwork    string
+		vnetResourceGroup string
+		link              bool
+		public            bool
+	}
+)
+
+func runMigrateAzureDNSMigrateCmd(cmd *cobra.Command, args []string) error {
+	if (azureMigrateOpts.zone == "" || azureMigrateOpts.oldResourceGroup == "") || (azureMigrateOpts.link == true && azureMigrateOpts.virtualNetwork == "") || (azureMigrateOpts.virtualNetwork != "" && azureMigrateOpts.vnetResourceGroup == "") {
+		return cmd.Help()
+	}
+
+	return azmigrate.DoMigrate(azureMigrateOpts.oldResourceGroup, azureMigrateOpts.newResourceGroup, azureMigrateOpts.zone, azureMigrateOpts.virtualNetwork, azureMigrateOpts.vnetResourceGroup, azureMigrateOpts.link, azureMigrateOpts.public)
+}
+
+func newMigrateAzureDNSMigrateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "migrate",
+		Short: "Migrate a zone",
+		RunE:  runMigrateAzureDNSMigrateCmd,
+	}
+
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.zone, "zone", "", "The zone to migrate")
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.oldResourceGroup, "oldResourceGroup", "", "The resource group of the legacy zone")
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.newResourceGroup, "newResourceGroup", "", "The resource group to create the private zone in")
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.virtualNetwork, "virtualNetwork", "", "The virtual network to create the private zone in")
+	cmd.PersistentFlags().StringVar(&azureMigrateOpts.vnetResourceGroup, "vnetResourceGroup", "", "The resource group the virtual network is in")
+	cmd.PersistentFlags().BoolVar(&azureMigrateOpts.link, "link", false, "Link the newly created private zone to the virtual network")
+	cmd.PersistentFlags().BoolVar(&azureMigrateOpts.public, "public", false, "Migrate a legacy public zone to a private zone")
+
+	return cmd
+}
+
+var (
+	azureMigrateTestOpts struct {
+		zone          string
+		resourceGroup string
+	}
+)
+
+func runMigrateAzureDNSTestCmd(cmd *cobra.Command, args []string) error {
+	if azureMigrateTestOpts.zone == "" || azureMigrateTestOpts.resourceGroup == "" {
+		return cmd.Help()
+	}
+
+	return azmigrate.DoTest(azureMigrateTestOpts.zone, azureMigrateTestOpts.resourceGroup)
+}
+
+func newMigrateAzureDNSTestCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "test",
```

Comment:
again, this seems like something that shouldn't be added to the cli

Reply:
Nuked!